### PR TITLE
Update quantity bundle items not working (GCOM-1176)

### PR DESCRIPTION
**Pull request by Jessevdpoel**

### CODE REVIEW

File: packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ export function TextFieldElement<TFieldValues extends FieldValues>({
   name,
   control,
   defaultValue,
+  castToNumber = true,
```
Comment (mikekeehnen) [inline packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx:25]: Why would we want to be able to manipulate this value via a prop? Looks to me like this should work out of the box

File: packages/magento-product-bundle/components/BundleProductOptions/BundleOptionValue.tsx [hunk 2]
```diff
@@ -86,10 +86,22 @@ export const BundleOptionValue = (props: ActionCardItemRenderProps<BundleOptionV
             size='small'
             label='Quantity'
             color={color}
+            castToNumber={false}
+            validation={{
+              pattern: { value: /^[1-9]|[1-9]d+$/, message: 'Quantity must be greater than 0' },
+            }}
+            InputProps={{ inputProps: { min: 1 } }}
```
Comment (mikekeehnen) [inline packages/magento-product-bundle/components/BundleProductOptions/BundleOptionValue.tsx:93]: We check if we have a number, but not cast it to a number?

File: packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx [hunk 3]
```diff
@@ -49,9 +50,10 @@ export function TextFieldElement<TFieldValues extends FieldValues>({
           name={name}
           value={value ?? ''}
           onChange={(ev) => {
-            onChange(
-              type === 'number' && ev.target.value ? Number(ev.target.value) : ev.target.value,
-            )
+            const fieldValue =
+              type === 'number' && ev.target.value ? Number(ev.target.value) : ev.target.value
+            const transformValue = transform ? transform(fieldValue) : fieldValue
```
Comment (paales) [inline packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx:55]: `transform` prop isn't used, so can be removed?

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx (hunk 3 above)
Comment (Jessevdpoel) [reply packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx:55]: transform prop is used by the BundleOptionValue component

File: packages/magento-product-bundle/components/BundleProductOptions/BundleOptionValue.tsx [hunk 4]
```diff
@@ -86,10 +86,22 @@ export const BundleOptionValue = (props: ActionCardItemRenderProps<BundleOptionV
             size='small'
             label='Quantity'
             color={color}
+            transform={String}
+            validation={{
+              pattern: { value: /^[1-9]|[1-9]d+$/, message: 'Quantity must be greater than 0' },
+            }}
```
[the remark this answers is not in the document]
Comment (Jessevdpoel) [reply packages/magento-product-bundle/components/BundleProductOptions/BundleOptionValue.tsx:92]: Lets discuss this.

**Pull request merged**